packages/dart_frog/test/src/response_test.dart

```diff
@@ -45,6 +56,28 @@ void main() {
       expect(response.headers['foo'], equals('bar,baz'));
     });
 
+    test('body can be read multiple times (sync)', () {
+      final body = json.encode({'test': 'body'});
+      final response = Response(body: body);
+
+      expect(response.body(), completion(equals(body)));
+      expect(response.body(), completion(equals(body)));
+
+      expect(response.json(), completion(equals(json.decode(body))));
+      expect(response.json(), completion(equals(json.decode(body))));
+    });
+
+    test('body can be read multiple times (async)', () async {
+      final body = json.encode({'test': 'body'});
+      final response = Response(body: body);
+
+      await expectLater(response.body(), completion(equals(body)));
+      await expectLater(response.body(), completion(equals(body)));
+
+      await expectLater(response.json(), completion(equals(json.decode(body))));
+      await expectLater(response.json(), completion(equals(json.decode(body))));
```

Comment:
this don't test that shelf if able to read the response and send it back, my guess is that will fail.

Reply:
Why not? Shelf just creates the request and response object just like we are doing here. I verified e2e that the changes do work. Feel free to try it yourself and let me know if you encounter any issues, thanks!